### PR TITLE
[Location sharing] - Declare background location permission in release manifest (PSF-1039)

### LINKED ISSUE #6158: [Location sharing] - Declare background location permission in release Manifest
#### What would you like to do?
Declare the background location permission inside the Release variant Manifest.

#### Why would you like to do it?
Since the live location feature can now be enabled via a labs settings, the background location permission should be declare in release mode.

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Moving the background location permission declaration from debug Manifest to release Manifest.

## Motivation and context

Closes #6158
Relates to #5635

## Tests

- Build the app in release mode
- Ensure the location sharing is enabled in Settings -> Preferences
- Ensure live location feature is enabled in Settings -> Labs

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: changelog.d/6158.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+[Live location sharing] - Declare the background location permission access into the release variant Manifest
```
Comment (onurays) [inline changelog.d/6158.misc:1]: It is funny to see similar PR & issue numbers.

**Review by onurays — APPROVED** [review]
I am excited to release this feature!

**Comment by mnaturel** [comment]
After some tests, I have a remark concerning the UX:
Even if I granted the microphone permission once, the dialog with checkbox to request permission appears again. I am wondering if we could directly request Android permissions without the extra dialog with checkboxes? It will be less painful for users.

**Review by ouchadam — CHANGES_REQUESTED** [review]
LGTM! _buttt_ to double check, has #5635 been completed? otherwise our release will be blocked :cry:

**Comment by Johennes** [comment]
> LGTM! _buttt_ to double check, has #5635 been completed? otherwise our release will be blocked 😢

The privacy policy update is under review in https://github.com/vector-im/policies/pull/29. I think it only needs typo fixes. @deniseal, do we already know when this is going to be deployed?

**Comment by mnaturel** [comment]
Closing this since the background location permission is not needed, see #6198

**Pull request closed without merging**